lib/tessel/deploy.js

```diff
@@ -68,44 +68,31 @@ Tessel.prototype.restartScript = function(opts) {
   var filepath = isPush ? PUSH_PATH : RUN_PATH;
 
   return new Promise(function(resolve, reject) {
-    return self.connection.exec(commands.readFile(filepath + opts.entryPoint))
-      .then(function(remoteProcess) {
-        var error = '';
-
-        remoteProcess.stderr.on('data', function(data) {
-          error += data.toString();
-        });
-
-        remoteProcess.once('close', function() {
-          if (error.length) {
-            if (error.indexOf('No such file or directory') !== -1) {
-              error = '"' + opts.entryPoint + '" not found on ' + self.name;
-            }
+    return self.simpleExec(commands.readFile(filepath + opts.entryPoint))
+      .then(function() {
+        if (isPush) {
+          // Start the script from flash memory
+          return actions.startPushedScript(self, opts.entryPoint)
+            .then(resolve).catch(reject);
+        } else {
+          // Start the script in RAM
+          return actions.runScript(self, filepath, opts.entryPoint)
+            .then(resolve).catch(reject);
+        }
+      })
+      .catch(function(error) {
+        if (error.message.indexOf('No such file or directory') !== -1) {
+          error = '"' + opts.entryPoint + '" not found on ' + self.name;
+        }
 
-            return reject(error);
-          } else {
-            if (isPush) {
-              // Start the script from flash memory
-              actions.startPushedScript(self, opts.entryPoint)
-                .then(resolve).catch(reject);
-            } else {
-              // Start the script in RAM
-              actions.runScript(self, filepath, opts.entryPoint)
-                .then(resolve).catch(reject);
-            }
-          }
-        });
+        return reject(error);
       });
   });
 };
 
 actions.execRemoteCommand = function(tessel, command, filepath) {
-  return new Promise(function(resolve) {
-    return tessel.connection.exec(commands[command](filepath))
-      .then(function(remoteProcess) {
-        remoteProcess.once('close', resolve);
-      });
-  });
+  return tessel.simpleExec(commands[command](filepath))
+    .catch(function() {});
```

Comment:
The previous implementation did so b/c it calls a command to stop the service that starts a tessel script on boot (if you `push` a project) regardless of whether the service has been started. If it hasn't been started, it will return an error. 

However, swallowing errors is almost never a good thing so I'll revise this a bit.

Good catch!